### PR TITLE
Added v3 token API containing metadata, including environment

### PR DESCRIPTION
**Pull request by evan-masseau**
# Description

- Replaces V2 identify API with V3 push tokens API call, including metadata
- Added collection of the debug/release environment 
- Collecting the push authorization state and background processing authorization 

# Check List

- [x] Are you changing anything with the public API? [NO]
- [x] Have you tested this change on real device?
- [x] Are your changes backwards compatible with previous SDK Versions?
- [x] Have you added unit test coverage for your changes?
- [x] Have you verified that your changes are compitable with all the operating system version this SDK currently supports?

# Manual Test Plan

1.

# Supporting Materials

### CODE REVIEW

File: Sources/KlaviyoSwift/AppContextInfo.swift [hunk 1]
```diff
@@ -30,17 +48,76 @@ struct AppContextInfo {
...
+
+    private func provisioningProfile() throws -> [String: Any]? {
+        guard let url = Bundle.main.url(forResource: "embedded", withExtension: "mobileprovision") else {
+            return nil
+        }
+
+        let binaryString = try String(contentsOf: url, encoding: .isoLatin1)
+
+        let scanner = Scanner(string: binaryString)
+        guard scanner.scanUpToString("<plist") != nil, let plistString = scanner.scanUpToString("</plist>"),
```
Comment (ndurell) [inline Sources/KlaviyoSwift/AppContextInfo.swift:115]: hmm

Comment (evan-masseau) [reply Sources/KlaviyoSwift/AppContextInfo.swift:115]: Yeah gotta test this on an ad hoc build to make sure it works as advertised

File: Sources/KlaviyoSwift/AppContextInfo.swift [hunk 2]
```diff
@@ -30,17 +48,76 @@ struct AppContextInfo {
...
+            #else
+            environment = "release"
+            #endif
+        }
+    }
+}
+
+extension UIDevice {
+    public enum PushEnvironment: String {
+        case unknown
```
Comment (jordan-griffin) [inline Sources/KlaviyoSwift/AppContextInfo.swift:89]: I know nothing on how this works, but I'm wondering why we would want an `unknown` environment. If APNS gives us nothing, can we assume thats a device running on prod and default to that?

Comment (evan-masseau) [reply Sources/KlaviyoSwift/AppContextInfo.swift:89]: i used this as a fallback to then use the DEBUG flag

Comment (ndurell) [reply Sources/KlaviyoSwift/AppContextInfo.swift:89]: I think this is indicating we can't determine the type.

Comment (jordan-griffin) [reply Sources/KlaviyoSwift/AppContextInfo.swift:89]: Aaah I see above. Thanks - makes perfect sense to me

Comment (ndurell) [reply Sources/KlaviyoSwift/AppContextInfo.swift:89]: Err the environment.

Comment (ndurell) [reply Sources/KlaviyoSwift/AppContextInfo.swift:89]: The question is what would we do with these tokens?

Comment (evan-masseau) [reply Sources/KlaviyoSwift/AppContextInfo.swift:89]: I suspect we'd only hit this if the app that doesn't use push notifications at all. In which case I'll just fall back on using the DEBUG compiler flag. 
As far as what we send the server I'll still only ever send `debug` or `release`

Comment (ndurell) [reply Sources/KlaviyoSwift/AppContextInfo.swift:89]: ok cool

File: Tests/KlaviyoSwiftTests/__Snapshots__/EncodableTests/testKlaviyoState.1.json [hunk 3]
```diff
@@ -7,17 +7,50 @@
...
+                "data" : {
+                  "attributes" : {
+                    "anonymous_id" : "foo",
+                    "email" : "foo",
+                    "phone_number" : "foo",
+                    "properties" : {
+
+                    }
+                  },
+                  "meta" : {
```
Comment (ndurell) [inline Tests/KlaviyoSwiftTests/__Snapshots__/EncodableTests/testKlaviyoState.1.json:40]: Ahh dang this part is different I think....

Comment (ndurell) [reply Tests/KlaviyoSwiftTests/__Snapshots__/EncodableTests/testKlaviyoState.1.json:40]: I'm not sure you can put meta in or not I mean

Comment (evan-masseau) [reply Tests/KlaviyoSwiftTests/__Snapshots__/EncodableTests/testKlaviyoState.1.json:40]: wondered about that, sad. it was nice to just re-use the same struct

File: Sources/KlaviyoSwift/InternalAPIModels.swift [hunk 4]
```diff
@@ -160,53 +178,105 @@ extension KlaviyoAPI.KlaviyoRequest {
...
-                let email: String?
-                let phoneNumber: String?
-                let externalId: String?
-                struct Append: Equatable, Codable {
-                    let pushToken: String
-                    enum CodingKeys: String, CodingKey {
-                        case pushToken = "$ios_tokens"
-                    }
-                }
+            var type = "push-token"
```
[the remark this answers is not in the document]
Comment (evan-masseau) [reply Sources/KlaviyoSwift/InternalAPIModels.swift:181]: Oh it gives a warning about how this property can't be decoded if declared as a let. Which i guess is "just" a warning but it makes me want to leave it a var.

File: Sources/KlaviyoSwift/AppContextInfo.swift (hunk 1 above)
Comment (evan-masseau) [reply Sources/KlaviyoSwift/AppContextInfo.swift:115]: Confirmed ad-hoc build -> "release" woot

**Comment by evan-masseau** [comment]
Do not merge -- currently points to an alpha revision of the V3 API.

**Comment by evan-masseau** [comment]
closing to consolidate PRs

**Pull request closed without merging**